### PR TITLE
Support continuation lines in properties files

### PR DESCRIPTION
**Pull request by knutwannheden**
Java `.properties` files can have continuation lines. The parser already supported this but the continuation lines were lost after parsing, meaning that after applying a recipe and serializing the LST again, the continuation lines got lost.

Adds a new `Properties.Entry#keySource` property and a new `Properties.Value#source` property to keep the original parsed source text, so that it can be serialized again.

Fixes: #2473

### CODE REVIEW

**Comment by knutwannheden** [comment]
The `rewrite-spring` recipe `UseTlsJdbcConnectionString` would also require updating. Should the original constructor be kept for compatibility?

**Review by traceyyoshima — CHANGES_REQUESTED** [review]
The PropertiesParserTest#escapedEndOfLine fails.

There are some examples of tests that check the continuation rules here:
https://github.com/openrewrite/rewrite/compare/properies_parser_with_continued_lines

I don't think we need to cover all the cases immediately, but we should at least cover continued keys. I recall that occurring in a code base.

**Comment by knutwannheden** [comment]
@traceyyoshima I will try to check those tests for keys with line continuations and if easy to fix also fix it. Otherwise, if this PR doesn't represent a regression in this area, I would be inclined to integrate it anyway as an enhancement and fix the remaining work in follow-ups.

**Comment by knutwannheden** [comment]
I implemented support for continuation lines in the property keys, but I still need to pull in all tests from the original PR. Also, I suspect there could still be an uncovered corner case, when there are continuation lines in the prefixes (key, equals, or value).

The current parser is terse, but quite hard to follow. Thinking about replacing it with a proper ANTLR parser or rewriting it to be more readable.

**Comment by traceyyoshima** [comment]
Optionally, we can store the raw text, including the continuations, and then revise the get method to return the trimmed value.
For printing, we'd return the raw text with an additional getRawText or something along those lines.

That will allow recipes to check and set the value, replacing the continued text. As well as allow users to add continuations in the raw text if desired.

**Comment by knutwannheden** [comment]
I am thinking it may make sense to only allow to set the raw value and thus have the unescaped value be a "view" of that (i.e. the getter would unescape it on the fly).

**Comment by traceyyoshima** [comment]
> I am thinking it may make sense to only allow to set the raw value and thus have the unescaped value be a "view" of that (i.e. the getter would unescape it on the fly).

Agreed, that is essentially what I had in mind as well. Rather than having two fields, have one field where the get method will return the unescaped value. That will allow downstream recipes to use the same get and with methods without any changes. Then an additional method will return the continued text.

**Comment by knutwannheden** [comment]
This PR was superseded by #3770 which now finally fixes this issue.
